morpheus/_lib/src/stages/inference_client_stage.cpp

```diff
@@ -169,22 +171,31 @@ mrc::coroutines::AsyncGenerator<std::shared_ptr<MultiResponseMessage>> Inference
             // TensorMap output_tensors;
             // buffer_map_t output_buffers;
 
-            auto lock = std::shared_lock(m_session_mutex);
+            if (m_session == nullptr)
+            {
+                auto lock = std::unique_lock(m_session_mutex);
+                if (m_session == nullptr)
+                {
+                    m_session = m_client->create_session();
+                }
+            }
```

Comment:
There is a potential race condition here where multiple threads are racing to create the session. Look into moving this into the catch handler.